cmd/bbolt/main.go

```diff
@@ -1103,17 +1103,33 @@ func (cmd *benchCommand) Run(args ...string) error {
 	db.NoSync = options.NoSync
 	defer db.Close()
 
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+
 	// Write to the database.
 	var writeResults BenchResults
+	var keys []nestedKey
+
+	if options.ReadMode == "rnd" {
+		keys = make([]nestedKey, 0, options.Iterations)
+	} else {
+		keys = nil
+	}
+
 	fmt.Fprintf(cmd.Stderr, "starting write benchmark.\n")
-	if err := cmd.runWrites(db, options, &writeResults); err != nil {
+	if err := cmd.runWrites(db, options, &writeResults, r, &keys); err != nil {
```

Comment:
I know it works, but I would suggest to avoid the extra level of indirection (which is a little clumsy and less readable). I suggest to update runWrites's signature something like below,

```
func (cmd *benchCommand) runWrites(db *bolt.DB, options *BenchOptions, results *BenchResults, r *rand.Rand, cacheKeys bool) ([]nestedKey, error) {
    ....
}
```

Note that in golang, a slice is actually a struct with three fields: cap, len and a pointer to an underline array.  Copying a slice doesn't copy the data.